apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/AbstractSpan.java

```diff
@@ -138,12 +144,28 @@ public T activate() {
         final ElasticApmTracer tracer = this.tracer;
         previouslyActive = tracer.getActive();
         tracer.setActive(this);
+        List<SpanListener> spanListeners = tracer.getSpanListeners();
+        for (int i = 0; i < spanListeners.size(); i++) {
+            try {
+                spanListeners.get(i).onActivate(this);
+            } catch (Throwable t) {
+                logger.warn("Exception while calling {}#onActivate", spanListeners.get(i).getClass().getSimpleName(), t);
```

Comment:
You shouldn't catch-log all throwables willy-nilly. It will catch OutOfMemoryError and other serious VirtualMachineError that should bubble up.
Some programs re-throw if it's a subclass of VirtualMachineError | ThreadDeath | InterruptedException | LinkageError | ControlThrowable, which seems like a reasonable approach.

Reply:
Thx for the suggestion. Ideally I would not catch Throwables at all, the problem is that `MethodHandle#invoke` throws a Throwable and that I need to handle that somehow.